src/Symfony/Component/Form/AbstractTypeExtension.php

```diff
@@ -45,4 +46,13 @@ public function finishView(FormView $view, FormInterface $form, array $options)
     public function configureOptions(OptionsResolver $resolver)
     {
     }
+
+    public function getExtendedType()
```

Comment:
@xabbuh following your [comment](https://github.com/symfony/symfony/pull/24530#issuecomment-336118044), how do you plan to deprecate this method now?

Reply:
What about a deprecation message + BC layer here:

https://github.com/symfony/symfony/blob/61da48799c2ce020a72915311d053ec4557d6923/src/Symfony/Component/Form/AbstractExtension.php#L178

here:
https://github.com/symfony/symfony/blob/61da48799c2ce020a72915311d053ec4557d6923/src/Symfony/Component/Form/FormFactoryBuilder.php#L103

and here:
https://github.com/symfony/symfony/blob/61da48799c2ce020a72915311d053ec4557d6923/src/Symfony/Component/Form/FormFactoryBuilder.php#L114

Reply:
IMO we no longer need to deprecate the method.

Reply:
To elaborate a bit more on this: The new interface extends the existing `FormTypeExtensionInterface`. If we do not deprecate anything, we know that the extension will have the `getExtendedType()` method. And if we extend the `AbstractTypeExtension` class, we even do not have to care about this method as it is then provided by the base class in case we implement the new interface.

Reply:
And if we do not implement the new interface, nothing changes at all as we still need to implement the `getExtendedType()` method like we had to do before.

Reply:
My concern about `getExtendedType()` in `5.0` is that we have two methods almost equals (both return the type to extend) but the new one is much better, it also allows us autoconfigure the type extension, great! So when will it `"getExtendedType()"` be used in `5.0` userland?

My point, we'll have two choice to return the type to extend for `5.0+`, the first one is a bad choice if the second one is better, good condition to deprecate it IMO.

Plus, if we keep `getExtendedType()` in `AbstractTypeExtension` forever, IDEs like PhpStorm will not be able to autodetect the missing method to implement anymore, this will be seen in the exception on complication time, worst DX then.

At the end: I'd like (for `5.0` o `6.0`) to see the static version in `FormTypeExtensionInterface` instead of the current `getExtendedType()`, is it possible in long term? I'm dreaming, probably a nightmare :)

Reply:
`getExtendedType()` will only there for BC concers. But I am not convinced it's really necessary to deprecate it if the code will still work fine. That way we would save maintainers some hassle with updating their code.

Reply:
@yceruto `getExtendedType()` is now deprecated in the interface

Reply:
With `getExtendedType` deprecated, all the calls that are still done to it need to use the new static method instead when it exists. E.g. in AbstractExtension that yceruto mentioned. And it should also trigger a deprecation message or is this done automatically?